### ISSUE TITLE
Train from scratch

### ISSUE THREAD

**Issue by haoshuai714** [gingsi/coot-videotext#10]
Hello:
    I started the training model in accordance with the guidance you provided, but I found that the network has not converged all the time. May I ask what the problem is? Is there something wrong with the config file provided?

**Comment by gingsi** [comment]
Hi, please answer the following questions and I can check the problem:

Describe the bug
A clear and concise description of what the bug is. (What does it mean it "has not converged all the time"?)

To Reproduce
Steps to reproduce the behavior. (What exactly are you doing, which model, which dataset, what is the exact command)

Expected behavior
A clear and concise description of what you expected to happen.

Screenshots
If applicable, add screenshots to help explain your problem. Post the training log somewhere.

System Info:

    OS: [e.g. Ubuntu 18.04]
    Python version [e.g. 3.8.6]
    PyTorch version [e.g. 1.7.0+cu11]

Additional context
Add any other context about the problem here.

Once you have given me all those details I can check again. Currently I do not have enough information to help you.

Best,

**Comment by haoshuai714** [comment]
Hi:
**1.Train Task:**
Table 2: Video-paragraph retrieval results on AcitvityNet-captions dataset (val1).
# train from scratch
python train.py config/anet_coot.yaml --cuda --log_dir runs/anet_coot

2.I want to test "Video-paragraph retrieval results on AcitvityNet-captions".

**3. Problem:**
I run "python train.py config/anet_coot.yaml --cuda --log_dir runs/anet_coot".
The loss function doesn't go down:
1129 140208 INFO ---------- Training epoch 28 ----------
1129 140212 INFO E28[  0/157] T 55.309m LR 1.000e-06 L 0.0392
1129 140217 INFO E28[ 10/157] T 55.379m LR 1.000e-06 L 0.0423
1129 140221 INFO E28[ 20/157] T 55.446m LR 1.000e-06 L 0.0394
1129 140225 INFO E28[ 30/157] T 55.515m LR 1.000e-06 L 0.0283
1129 140229 INFO E28[ 40/157] T 55.581m LR 1.000e-06 L 0.0312
1129 140233 INFO E28[ 50/157] T 55.648m LR 1.000e-06 L 0.0270
1129 140237 INFO E28[ 60/157] T 55.715m LR 1.000e-06 L 0.0396
1129 140241 INFO E28[ 70/157] T 55.790m LR 1.000e-06 L 0.0454
1129 140245 INFO E28[ 80/157] T 55.855m LR 1.000e-06 L 0.0328
1129 140249 INFO E28[ 90/157] T 55.920m LR 1.000e-06 L 0.0388
1129 140253 INFO E28[100/157] T 55.984m LR 1.000e-06 L 0.0318
1129 140257 INFO E28[110/157] T 56.050m LR 1.000e-06 L 0.0465
1129 140301 INFO E28[120/157] T 56.115m LR 1.000e-06 L 0.0337
1129 140305 INFO E28[130/157] T 56.183m LR 1.000e-06 L 0.0288
1129 140309 INFO E28[140/157] T 56.252m LR 1.000e-06 L 0.0443
1129 140312 INFO E28[150/157] T 56.306m LR 1.000e-06 L 0.0268
1129 140314 INFO ---------- Validating epoch 28 ----------
ids ['v_--1DO2V4K74', 'v_--6bJUbfpnQ', 'v_-01K1HxqPB8', 'v_-02DygXbn6w']...
1129 140319 INFO Val [0/77] Loss 0.0407
1129 140336 INFO Val [10/77] Loss 0.0284
1129 140338 INFO Val [20/77] Loss 0.0352
1129 140341 INFO Val [30/77] Loss 0.0340
1129 140343 INFO Val [40/77] Loss 0.0346
1129 140346 INFO Val [50/77] Loss 0.0310
1129 140349 INFO Val [60/77] Loss 0.0302
1129 140352 INFO Val [70/77] Loss 0.0264
1129 140358 INFO Retriev | R@1   | R@5   | R@10  | R@50  | MeanR |  MedR |    Sum
1129 140358 INFO Par2Vid | 0.601 | 0.871 | 0.932 | 0.985 |   1.0 |   5.2 |  2.457
1129 140358 INFO Vid2Par | 0.617 | 0.880 | 0.935 | 0.987 |   1.0 |   5.1 |  2.485
1129 140358 INFO Retrieval done: runs/anet_coot 4917 Items.
**4. Question:**
The Recall Sum on the val_1 is always around 2.

**Comment by gingsi** [comment]
This looks perfectly fine, you are getting the results as promised in the table (~61% retrieval @ 1)

We don't care about training loss going to 0 (which would mean overfitting), we want good validation results.

I don't see the problem, please explain more what you expect us to do.

**Comment by haoshuai714** [comment]
Hello:
   I train the model as your GitHub guide, then test in the test dataset. The test result, as follow:
1130 012718 INFO Retriev | R@1   | R@5   | R@10  | R@50  | MeanR |  MedR |    Sum
1130 012718 INFO Par2Vid | 0.606 | 0.870 | 0.932 | 0.984 |   1.0 |   5.1 |  2.459
1130 012718 INFO Vid2Par | 0.616 | 0.880 | 0.935 | 0.987 |   1.0 |   5.0 |  2.48

Can you provide the anet_coot.yaml file for your training model?

**Comment by gingsi** [comment]
The anet_coot.yaml that is already uploaded is the best model from the paper.

The retrieval results you are showing are fine.

I still have no idea what you are asking for

**Comment by haoshuai714** [comment]
I test "Video-paragraph retrieval results on AcitvityNet-captions".
But the result in the test dataset as follow:
1130 012718 INFO Retriev | R@1 | R@5 | R@10 | R@50 | MeanR | MedR | Sum
1130 012718 INFO Par2Vid | 0.606 | 0.870 | 0.932 | 0.984 | 1.0 | 5.1 | 2.459
1130 012718 INFO Vid2Par | 0.616 | 0.880 | 0.935 | 0.987 | 1.0 | 5.0 | 2.48
That's a big difference from the 61% in the paper!
While this code works, the results are pretty bad.
What's the problem, please?

**Comment by haoshuai714** [comment]
I got it wrong. R@1=0.606, representing 60.6%.Thank you for your patience

**Comment by gingsi** [comment]
Hi, you are welcome. Glad we could resolve it.
